client/reader/following-manage/subscriptions.jsx

```diff
@@ -49,14 +54,37 @@ class FollowingManageSubscriptions extends Component {
 		} );
 	}
 
+	sortFollows( follows, sortOrder ) {
+		const { getFeed, getSite } = this.props;
+
+		if ( sortOrder === 'alpha' ) {
+			return sortBy( follows, follow => {
+				const feed = getFeed( follow.feed_ID );
+				const site = getSite( follow.site_ID );
+				const displayUrl = formatUrlForDisplay( follow.URL );
+				return trimStart( getFeedTitle( site, feed, displayUrl ).toLowerCase() );
+			} );
+		}
+
+		return reverse( sortBy( follows, [ 'date_subscribed' ] ) );
+	}
+
+	handleSortChange = ( sort ) => {
+		page.replace( addQueryArgs( { sort }, window.location.pathname + window.location.search ) );
+	};
+
 	componentWillReceiveProps( nextProps ) {
-		const forceRefresh = ( nextProps.query !== this.props.query );
+		const forceRefresh = (
+			( nextProps.query !== this.props.query ) ||
+			( nextProps.sortOrder !== this.props.sortOrder )
+		);
 		this.setState( { forceRefresh } );
 	}
 
 	render() {
-		const { follows, width, translate, query, followsCount } = this.props;
+		const { follows, width, translate, query, followsCount, sortOrder } = this.props;
 		const filteredFollows = this.filterFollowsByQuery( query );
+		const sortedFollows = this.sortFollows( filteredFollows, sortOrder );
```

Comment:
might consider lifting this up to component state. right now we're sorting the array on every render... unsure how many wasted renders that ends up generating though. Doesn't feel noticeable so far even with > 1K follows.